plc4j/utils/opm/src/main/java/org/apache/plc4x/java/opm/PlcEntityManager.java

```diff
@@ -0,0 +1,380 @@
+        this.driverManager = driverManager;
+    }
+
+    public <T> T read(Class<T> clazz) throws OPMException {
+        PlcEntity annotation = getPlcEntityAndCheckPreconditions(clazz);
+        String source = annotation.value();
+
+        try (PlcConnection connection = driverManager.getConnection(source)) {
+
+            if (!connection.getReader().isPresent()) {
+                throw new OPMException("Unable to get Reader for connection with url '" + source + "'");
+            }
+
+            PlcReader reader = connection.getReader().get();
+
+            PlcReadRequest.Builder requestBuilder = reader.readRequestBuilder();
+
+            // Do the necessary queries for all fields
+            // HashMap<ReadRequestItem<?>, Field> requestItems = new HashMap<>();
+            for (Field field : clazz.getDeclaredFields()) {
+                PlcField fieldAnnotation = field.getAnnotation(PlcField.class);
+                if (fieldAnnotation == null) {
+                    // Ignore that field
+                    continue;
+                }
+                // Create the suitable Request
+                String query = fieldAnnotation.value();
+                requestBuilder.addItem(field.getName(), query);
+            }
+
+            // Build the request
+            PlcReadRequest request;
+            try {
+                request = requestBuilder.build();
+            } catch (PlcInvalidFieldException e) {
+                throw new OPMException("Unable to parse one field request", e);
+            }
+
+            // Perform the request
+            PlcReadResponse<?> response;
+            try {
+                // TODO: make configurable.
+                response = reader.read(request).get(1_000, TimeUnit.MILLISECONDS);
+            } catch (InterruptedException | ExecutionException e) {
+                throw new OPMException("Request fetching not able", e);
+            } catch (TimeoutException e) {
+                throw new OPMException("Timeout during fetching values", e);
+            }
+
+            // Construct the Object
+            T instance = clazz.getConstructor().newInstance();
+
+            // Fill all requested fields
+            for (String fieldName : response.getFieldNames()) {
+                setField(clazz, instance, response, fieldName);
+            }
+            return instance;
+        } catch (PlcConnectionException e) {
+            throw new OPMException("Unable to get connection with url '" + source + "'", e);
+        } catch (Exception e) {
+            throw new OPMException("Unable to fetch PlcEntity " + clazz.getName(), e);
+        }
+    }
+
+    /**
+     * Returns a connected proxy.
+     *
+     * @param clazz clazz to be connected.
+     * @param <T>   type of param {@code clazz}.
+     * @return a connected entity.
+     * @throws OPMException when proxy can't be build.
+     */
+    public <T> T connect(Class<T> clazz) throws OPMException {
```

Comment:
confuses me a bit. Is this the same as merge in jpa?

Reply:
Its kinda similar to `find(...)` in JPA because it returns a connected entity.
But as it does not have a primary key (makes no sense) I decided to name it connect.
But thinking about it, it is pretty similar to `merge(...)` with the difference that it takes a class and not an instance.
But I'm open for changes in the naming.